CA_DataUploaderLib/HeatingController.cs

```diff
@@ -366,6 +407,23 @@ internal void MakeDecision(Events e, double data = 0)
                             case (States.Off or States.Running, Events.oven or Events.ovenarea):
                                 target = data;
                                 break;
+                            case (States.Off or States.Running, Events.pgain):
+                                pgain = Math.Min(data, _config.OvenProportionalControlUpdatesConf!.MaxProportionalGain);
+                                break;
+                            case (States.Off or States.Running, Events.controlperiodseconds):
+                                controlperiodseconds = Math.Min(data, _config.OvenProportionalControlUpdatesConf!.MaxControlPeriod.TotalSeconds);
+                                break;
+                            case (States.Off or States.Running, Events.maxoutputs):
+                                maxoutput = Math.Min(data / 100, _config.OvenProportionalControlUpdatesConf!.MaxOutputPercentage);
+                                break;
+                            case (States.Off or States.Running, Events.vector):
```

Comment:
is this case necessary?

Reply:
yes, without it a plugin can update the values outside the range and the heater decision would end up using those. Removing it breaks test:PControlUsesUsesMaxConfigurationForVectorValuesAboveAllowedRange

It does relies on the order of plugins, where the oven area decision runs before the heater decisions. Additionally built in decisions run before any plugin, so it is always the value within range that is used. 

Note I did not do it this way for the minimum valid values, so that the fix for that issue applies to both the new feature and when using values from the configuration file.